app/models/case.js

```diff
@@ -26,6 +29,23 @@ export default Model.extend({
     return A(this.get('pieces').toArray()).sort((pieceA, pieceB) => compareFunction(new VRDocumentName(pieceA.get('name')), new VRDocumentName(pieceB.get('name'))));
   }),
 
+  // Computed.
+  hasBvr: computed('pieces.@each', function() {
+    const _this = this;
+    return (async() => {
+      const documentTypeBesluit = await _this.store.findRecord('document-type', CONFIG.documentType.besluitVlaamseRegering.id);
+      return await _this.caseService.hasPieceOfType(_this, documentTypeBesluit);
+    })(_this);
+  }),
+
+  hasDecreet: computed('pieces.@each', function() {
+    const _this = this;
+    return (async() => {
+      const documentTypeDecreet = await _this.store.findRecord('document-type', CONFIG.documentType.decreet.id);
+      return await _this.caseService.hasPieceOfType(_this, documentTypeDecreet);
+    })(_this);
+  }),
+
```

Comment:
Kan dit niet gewoon geschreven worden als een async function? Leest wat gemakkelijker en in een computed zou het mogelijk moeten zijn. Ik zou enkel een getter in octane components op deze manier schrijven.. maar dit is misschien persoonlijke flavor

Reply:
Aangepast